crates/bevy_render/src/lib.rs

```diff
@@ -236,3 +242,48 @@ fn check_for_render_resource_context(context: Option<Res<Box<dyn RenderResourceC
         );
     }
 }
+
+fn window_icon_changed(
+    mut map: Local<HashMap<WindowId, Handle<Texture>>>,
+    textures: Res<Assets<Texture>>,
+    mut windows: ResMut<Windows>,
+    asset_server: Res<AssetServer>,
+) {
+    for window in windows.iter_mut() {
+        /* Insert new icon changed */
+        if let Some(WindowIcon::Path(path)) = window.icon() {
+            if let Some(handle) = map.get(&window.id()) {
+                if let Some(handle_path) = asset_server.get_handle_path(handle) {
+                    if handle_path.path() != path {
+                        map.insert(window.id(), asset_server.load(path.clone()));
+                    }
+                }
+            } else {
+                map.insert(window.id(), asset_server.load(path.clone()));
+            }
```

Comment:
```suggestion
            match map.entry(window.id()) {
                Entry::Occupied(mut o) => {
                    let handle_path = asset_server.get_handle_path(o.get()).unwrap();
                    if handle_path != path {
                        o.insert(asset_server.load(path.clone());
                    }
                }
                Entry::Vacant(v) => v.insert(asset_server.load(path.clone()),
            }
```

Generally, it's preferred to use the `Entry` API when dealing when maps.
Also, I'm pretty sure you can call `unwrap()` on the `get_handle_path` since we know that if that handle is in our `Local` map, then we've already added it to the `AssetServer`.

Reply:
Thanks for the hint. 
About the unwrap -> this leads to an immediate panic. Not sure why. 
Maybe someone with more experience with bevy_asset can comment on that.

Reply:
Ahhh yes, ok that makes sense. 
We add the handle into the map as we're loading the asset. 
The asset doesn't have to be laoded yet so we do need to do 
```rust
if let Some(handle_path) = asset_server.get_handle_path(o.get())
```

Reply:
So if 'get_handle_path' only return a path, when the asset is loaded, then this part only makes sense if someone manages to issue a path and then another path before the loading. 
I guess I does not hurt to have this here, but it is unlikely to be triggered under normal conditions

Reply:
Ok so apparently, `get_handle_path` will always return `None`. See #1290 for more details haha :)

Reply:
#2310 was merged so `get_handle_path` should now be working (once you rebase on main)!!!

Reply:
I was able to confirm with info! that it will return an actual path now :)